component-runtime-testing/component-runtime-junit/src/main/java/org/talend/sdk/component/maven/MavenDecrypter.java

```diff
@@ -82,21 +100,63 @@ public Server find(final String serverId) {
         }
 
         final MvnServerExtractor extractor = new MvnServerExtractor(master, serverId);
-        try (final InputStream is = new FileInputStream(settings)) {
-            parser.parse(is, extractor);
-        } catch (final IOException | SAXException e) {
-            throw new IllegalArgumentException(e);
+        for (final File file : settings) {
+            try (final InputStream is = new FileInputStream(file)) {
+                parser.parse(is, extractor);
+            } catch (final IOException | SAXException e) {
+                throw new IllegalArgumentException(e);
+            }
+            if (extractor.server != null) {
+                return extractor.server;
+            }
         }
-        if (extractor.server == null) {
-            throw new IllegalArgumentException("Didn't find " + serverId + " in " + settings);
+
+        throw new IllegalArgumentException("Didn't find " + serverId + " in " + settings);
+    }
+
+    private static SAXParser newSaxParser() {
+        final SAXParserFactory factory = SAXParserFactory.newInstance();
+        factory.setNamespaceAware(false);
+        factory.setValidating(false);
+        try {
+            factory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+            factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, Boolean.TRUE);
+        } catch (final ParserConfigurationException | SAXNotRecognizedException | SAXNotSupportedException ex) {
+            // ignore
         }
-        return extractor.server;
+
+        final SAXParser parser;
+        try {
+            parser = factory.newSAXParser();
+        } catch (final ParserConfigurationException | SAXException e) {
+            throw new IllegalStateException(e);
+        }
+
+        try {
+            parser.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, "");
+            parser.setProperty(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
+        } catch (final SAXNotRecognizedException | SAXNotSupportedException e) {
+            // ignore
+        }
+
+        return parser;
     }
 
     private static File getM2() {
         return ofNullable(System.getProperty("talend.maven.decrypter.m2.location"))
                 .map(File::new)
-                .orElseGet(() -> new File(System.getProperty("user.home"), ".m2"));
+                .orElseGet(() -> new File(System.getProperty(USER_HOME), ".m2"));
+    }
+
+    private static List<File> findSettingsFiles() {
+        return Stream.of(
+                new File(getM2(), FILE_SETTINGS),
+                ofNullable(System.getenv(M2_HOME))
+                        .map(File::new)
+                        .map(it -> new File(it, "conf/" + FILE_SETTINGS))
```

Comment:
AFAIR, `conf/` is for maven home instead. Not sure it's in m2 home.

Reply:
yep, confirmed https://maven.apache.org/settings.html

Reply:
yes, and M2_HOME it's a path to maven installation directory
https://www.baeldung.com/java-maven-environment-variables
for maven 1 it was MAVEN_HOME
for maven 2 they renamed in to M2_HOME
what is for maven 3 - I don't know

Reply:
Okay, I see in that explanation, that they start to call it again MAVEN_HOME
I can change it to MAVEN_HOME
or we can support both
What do you think?

Reply:
As we handle multiple locations, let's put maven home as latest one.
Few chances that user uses that one...